analyzer/consensus/consensus.go

```diff
@@ -536,6 +511,124 @@ func (m *Main) queueNodeEvents(batch *storage.QueryBatch, data *storage.Registry
 	return nil
 }
 
+func (m *Main) queueRegistryEvents(batch *storage.QueryBatch, data *storage.RegistryData) error {
+	eventInsertQuery := m.qf.ConsensusEventInsertQuery()
+	backend := analyzer.BackendRegistry
+
+	for _, event := range data.Events {
+		var ty analyzer.Event
+		var body []byte
+		var err error
+
+		switch {
+		case event.RuntimeEvent != nil:
+			ty = analyzer.EventRegistryRuntime
+			body, err = json.Marshal(event.RuntimeEvent)
+
+			if err != nil {
+				return err
+			}
+		case event.EntityEvent != nil:
+			ty = analyzer.EventRegistryEntity
+			body, err = json.Marshal(event.EntityEvent)
+
+			if err != nil {
+				return err
+			}
+		case event.NodeEvent != nil:
+			ty = analyzer.EventRegistryNode
+			body, err = json.Marshal(event.NodeEvent)
+
+			if err != nil {
+				return err
+			}
+		case event.NodeUnfrozenEvent != nil:
+			ty = analyzer.EventRegistryNodeUnfrozen
+			body, err = json.Marshal(event.NodeUnfrozenEvent)
+
+			if err != nil {
+				return err
+			}
+		}
+
+		batch.Queue(eventInsertQuery,
+			backend.String(),
+			ty.String(),
+			string(body),
+			data.Height,
+			event.TxHash.Hex(),
+			0,
```

Comment:
Probably comment that this is gonna be `0` because event should be only emitted once per transaction.

Reply:
Actually, let's remove this. I think we can still achieve a reasonable amount of reference just from `txn_block` and `txn_hash`.

Reply:
Not doing anymore